artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -649,9 +647,7 @@ public AddressInfo removeAddressInfo(SimpleString address, boolean force) throws
 
    @Override
    public AddressInfo getAddressInfo(SimpleString addressName) {
-      synchronized (addressLock) {
-         return addressManager.getAddressInfo(addressName);
-      }
+      return addressManager.getAddressInfo(addressName);
```

Comment:
There's no need to lock here?

Reply:
@jbertram What do you think about this?

Reply:
I don't see a need for the `synchronized (addressLock)` given that the underlying data structure in the address manager is a `ConcurrentHashMap` and no other operations are being done (unlike in the other places which use `addressLock`). I think this fix looks good.

Reply:
@jbertram thanks.. that confirms what I thought.

I'm running the whole testsuite to confirm it.